### PR TITLE
API Compat tool in ML.NET

### LINKED ISSUE #3602: Need to add API breaking change definition and enforce it
With version `1.0.0` release we need to formally adopt the API breaking change definitions similar to [CoreFx](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-definitions.md) and [here](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-changes.md). We also need to add a test to run [APICompat](https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.ApiCompat) tool to enforce these rules. @eerhardt @TomFinley @glebuk

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #3602.

We need to ensure that future changes to ML.NET will not break the stable API released in 1.0.0.

This PR introduces the [API Compat](https://github.com/dotnet/arcade/tree/e5c20e707876cb9596d1cbcc996f80c25698fdfe/src/Microsoft.DotNet.ApiCompat) tool from dotnet/Arcade. The API Compat tool runs as part of the build process and compares the assemblies with those found in the stable nugets referenced in the `Microsoft.ML.StableAPI` project.

The tool is only run for the assemblies which will be part of the stable nugets. Here is a list of those assemblies and the relative nugets:

| Stable Nuget | Stable Assemblies|
|--|--|
|Microsoft.ML| Microsoft.ML.Core, Microsoft.ML.Data, Microsoft.ML.KMeansClustering, Microsoft.ML.PCA, Microsoft.ML.StandardTrainers, MIcrosoft.ML.Transforms, Microsoft.ML.Analyzer|
|Microsoft.ML.DataView|  Microsoft.ML.DataView|
|Microsoft.ML.CpuMath |  Microsoft.ML.CpuMath|
|Microsoft.ML.FastTree |  Microsoft.ML.FastTree|
|Microsoft.ML.LightGbm |  Microsoft.ML.LightGbm|
|Microsoft.ML.ImageAnalytics |  Microsoft.ML.ImageAnalytics|
|Microsoft.ML.MklComponents |  Microsoft.ML.Mkl.Components|

Note: the tool does not run on Microsoft.ML.Analyzer as it does not have a public API, so no need to check backwards compatibility.

Still to do:
- [x] Fix possible bug in API Compat on handling of Attributes.
- [x] ~~Add a build setting to disable the API Compat tool from command line. This will allow people to work on breaking changes.~~ Not needed see comments below.
- [x] Update the version of the ML.NET nugets to 1.0.0 when available

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -124,4 +124,14 @@
...
   </PropertyGroup>
+  
+  <!-- API Compat -->
+  <PropertyGroup>
+    <!-- only validate that the current implementation is compatible
+         with the old one, not vice-versa (since latest may have new
+         API missing from old) -->
+    <RunApiCompatForSrc>true</RunApiCompatForSrc>
+    <RunMatchingRefApiCompat>false</RunMatchingRefApiCompat>
+    <MicrosoftDotNetApiCompatPackageVersion>1.0.0-beta.19225.5</MicrosoftDotNetApiCompatPackageVersion>
```
Comment (ericstj) [inline Directory.Build.props]: This property should be moved to https://github.com/dotnet/machinelearning/blob/master/build/Dependencies.props

File: src/Directory.Build.targets [hunk 2]
```diff
@@ -26,4 +26,29 @@
...
 
+  <!-- API Compat -->
+  <PropertyGroup Condition="'$(RunApiCompat)' == 'true'">
+    <!-- resolve contract for APICompat as part of resolve references -->
+    <ResolveReferencesDependsOn>
+      $(ResolveReferencesDependsOn);
+      ResolveMatchingContract
+    </ResolveReferencesDependsOn>
+    
+    <StabelApiProject>$(RepoRoot)tools-local\Microsoft.ML.StableApi\Microsoft.ML.StableApi.csproj</StabelApiProject>
```
Comment (ericstj) [inline src/Directory.Build.targets]: nit: spelling

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  <!-- The purpose of this target is to return a path from a referenced
+       package / project -->
+  <Target Name="GetContract"
+          DependsOnTargets="ResolveReferences"
+          Returns="@(_contractReferencePath)">
+    <Error Condition="'$(ContractName)' == ''" Text="ContractName must be specified" />
+    <ItemGroup>
+      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
+    </ItemGroup>
+    <Error Condition="'$(ContractName)' == ''" Text="Could not locate $(ContractName)" />
```
Comment (ericstj) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: This should be `'@(_contractReferencePath)' == ''`.

File: src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj [hunk 4]
```diff
@@ -1,8 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard1.3</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
     <IncludeAnalyzerInPackage>Microsoft.ML</IncludeAnalyzerInPackage>
+    <RunApiCompat>true</RunApiCompat>
```
Comment (ericstj) [inline src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj]: Do you actually ship this analyzer?  Also changing targetFramework is a breaking change since it drops compatibility with what you previously targeted.

File: src/Directory.Build.targets [hunk 5]
```diff
@@ -26,4 +26,29 @@
 
   </Target>
 
+  <!-- API Compat -->
```
Comment (ericstj) [inline src/Directory.Build.targets]: Consider moving these to a seperate targets file if that is a convention you'd like to follow in this repo.

**Comment by ericstj** [comment]
>  Add a build setting to disable the API Compat tool from command line. This will allow people to work on breaking changes.

The right way to do this is to build once specifying /p:BaselineAllAPICompatError=true.  This will generate text files in the repo that list all the errors and let the build succeed despite the compatibility issues.  Typically you then file issues to address all the baselines by the time you ship the next version.

File: src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj (hunk 4 above)
Comment (eerhardt) [reply src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj]: Yes we ship this analyzer.

But this project has no public APIs, so I don't believe we need to Run API Compat on it.

---
In reply to: [279938491](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279938491) [](ancestors = 279938491)

Comment (artidoro) [reply src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj]: Should the targetFramework remain 1.3 @eric.erhardt@microsoft.com?
I'll remove the `RunApiCompat`.

---
In reply to: [279940309](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279940309) [](ancestors = 279940309,279938491)

File: src/Microsoft.ML.Core/Microsoft.ML.Core.csproj [hunk 6]
```diff
@@ -6,11 +6,11 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DefineConstants>CORECLR</DefineConstants>
     <RootNamespace>Microsoft.ML</RootNamespace>
+    <RunApiCompat>true</RunApiCompat>
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Microsoft.ML.Core.csproj]: Can we instead set `RunApiCompat` based on if `IsStableProject` is set to true?

[https://github.com/dotnet/machinelearning/blob/master/build/BranchInfo.props#L2-L23](https://github.com/dotnet/machinelearning/blob/master/build/BranchInfo.props#L2-L23)

In the root Directory.Build.props, you should be able to say:

```xml
  <PropertyGroup>
    <RunApiCompat Condition="'$(RunApiCompat)` == '' AND '$(IsStableProject)' == 'true'">true</RunApiCompat>
  </PropertyGroup>
```

or simply:

```xml
  <PropertyGroup>
    <RunApiCompat Condition="'$(RunApiCompat)` == ''">$(IsStableProject)</RunApiCompat>
  </PropertyGroup>
```

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 7]
```diff
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <!-- needs to contain all frameworks which src projects wish to restore -->
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.ML" Version="1.0.0-preview-27625-16"/>
```
Comment (eerhardt) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: (nit) you can use the latest 1.0.0-preview build: `1.0.0-preview-27630-5`. We just spun it and will have the official `1.0.0` up soon.

File: src/Directory.Build.targets (hunk 5 above)
Comment (artidoro) [reply src/Directory.Build.targets]: I think we only have one for the src repo. But let me know if there is a better way.

---
In reply to: [279938766](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279938766) [](ancestors = 279938766)

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 7 above)
Comment (artidoro) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Will update to the new build and change to 1.0.0 as soon as it is available.

---
In reply to: [279941849](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279941849) [](ancestors = 279941849)

File: src/Microsoft.ML.Core/Microsoft.ML.Core.csproj (hunk 6 above)
Comment (artidoro) [reply src/Microsoft.ML.Core/Microsoft.ML.Core.csproj]: Sure that seems to represent more clearly the logic, I'll do it.

---
In reply to: [279941488](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279941488) [](ancestors = 279941488)

**Comment by ericstj** [comment]
```
D:\a\1\s\packages\microsoft.dotnet.apicompat\1.0.0-beta.19225.5\build\Microsoft.DotNet.ApiCompat.targets(72,5): error : CannotChangeAttribute : Attribute 'System.AttributeUsageAttribute' on 'Microsoft.ML.ExtensionBaseAttribute' changed from '[AttributeUsageAttribute(4)]' in the contract to '[AttributeUsageAttribute(AttributeTargets.Class)]' in the implementation. [D:\a\1\s\src\Microsoft.ML.Core\Microsoft.ML.Core.csproj]
```

This is happening because APICompat is not resolving the System.AttributeTargets enum from the contract.   It's not resolving because you aren't providing the dependencies.  Today these can be passed in via the `$(ContractOutputPath)` property but that isn't very sensible.  I'll submit a change to APICompat to make that easier.
https://github.com/dotnet/arcade/blob/ac8d88df02d246d3147338fcfb03b1b93dc84b53/src/Microsoft.DotNet.ApiCompat/build/Microsoft.DotNet.ApiCompat.targets#L46-L52

**Comment by artidoro** [comment]
That would be great! Thank you

---
In reply to: [488135991](https://github.com/dotnet/machinelearning/pull/3623#issuecomment-488135991) [](ancestors = 488135991)

**Comment by artidoro** [comment]
> > Add a build setting to disable the API Compat tool from command line. This will allow people to work on breaking changes.
> 
> The right way to do this is to build once specifying /p:BaselineAllAPICompatError=true. This will generate text files in the repo that list all the errors and let the build succeed despite the compatibility issues. Typically you then file issues to address all the baselines by the time you ship the next version.

@ericstj is it possible to specify the path for the generated file?
Also, how would we specify known diffs going forward? I would like to write some documentation on how to do that.

**Comment by ericstj** [comment]
by default the generated file gets written to the same directory as the project with the name, but you can customize it: https://github.com/dotnet/arcade/blob/ac8d88df02d246d3147338fcfb03b1b93dc84b53/src/Microsoft.DotNet.ApiCompat/build/Microsoft.DotNet.ApiCompat.targets#L22-L23

> how would we specify known diffs going forward?

You can use the same option I mentioned when building individual projects, you can also copy the errors and paste them into the baseline file.  I want to stress that these aren't "known diffs" these are compatibility bugs that break your customers and you really shouldn't be baselining them or introducing the suppressions.

**Comment by ericstj** [comment]
To provide paths to contract dependencies, add the following in your `GetContract` target:
```
      <_allReferenceDirectories Include="%(ReferencePath.RootDir)%(ReferencePath.Directory)" />
      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
      <_contractReferencePath DependencyPaths="@(_allReferenceDirectories)" />
```

Then in `ResolveMatchingContract` add the following after you get the ResolvedMatchingContract item:
```
<PropertyGroup>
  <ContractOutputPath>%(ResolvedMatchingContract.DependencyPaths)</ContractOutputPath>
</PropertyGroup>
```

I'm putting a feature into the APICompat targets that will make the latter property unnecessary.

File: src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj (hunk 4 above)
Comment (eerhardt) [reply src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj]: I wouldn't change the TFM unless absolutely necessary. I think we should be able to revert this file.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 7 above)
Comment (eerhardt) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: It's available on myget now. You should be able to use `1.0.0`.

File: src/Microsoft.ML.Core/Microsoft.ML.Core.csproj [hunk 8]
```diff
@@ -10,7 +10,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.ML.DataView\Microsoft.ML.DataView.csproj" />
-    
+
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Microsoft.ML.Core.csproj]: I think you can revert all these .csproj files. No need cluttering the history making unnecessary changes.

File: src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj [hunk 9]
```diff
@@ -3,6 +3,9 @@
   <PropertyGroup>
     <TargetFramework>netstandard1.3</TargetFramework>
     <IncludeAnalyzerInPackage>Microsoft.ML</IncludeAnalyzerInPackage>
+
+    <!-- No public surface in this assembly -->
+    <RunApiCompat>false</RunApiCompat>
```
Comment (eerhardt) [inline src/Microsoft.ML.Analyzer/Microsoft.ML.Analyzer.csproj]: This shouldn't be necessary since ML.Analyzer isn't listed as a stable project. Can be reverted.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.ML" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.DataView" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.CpuMath" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.FastTree" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.LightGbm" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.ImageAnalytics" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.Mkl.Components" Version="1.0.0-preview-27630-5"/>
+    <PackageReference Include="Microsoft.ML.Mkl.Redist" Version="1.0.0-preview-27630-5"/>
```
Comment (eerhardt) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: The Redist package doesn't have public APIs. Do we need to include it here?

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 11]
```diff
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (eerhardt) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: This project will actually build a `.dll` when we build the `.sln` file. That seems unnecessary. Maybe we should override the `Build` target or maybe not name it `.csproj` and instead just `.proj`? It's mission in life isn't to build `.cs` files into a `.dll`, but instead pull down external packages and provide the `GetContract` target.

File: Directory.Build.props [hunk 12]
```diff
@@ -124,4 +124,13 @@
   <PropertyGroup>
     <CustomAfterMicrosoftCommonTargets>$(RepoRoot)build\AfterCommonTargets.targets</CustomAfterMicrosoftCommonTargets>
   </PropertyGroup>
+  
+  <!-- API Compat -->
```
Comment (eerhardt) [inline Directory.Build.props]: Can these properties go into `src\Directory.Build.props` instead? Grouping them with the `RunApiCompat` property would make sense IMO.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 10 above)
Comment (artidoro) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Yes you are right I'll remove that.

---
In reply to: [280104238](https://github.com/dotnet/machinelearning/pull/3623#discussion_r280104238) [](ancestors = 280104238)

File: Directory.Build.props (hunk 12 above)
Comment (artidoro) [reply Directory.Build.props]: Yes I'll do that!

---
In reply to: [280106010](https://github.com/dotnet/machinelearning/pull/3623#discussion_r280106010) [](ancestors = 280106010)

**Comment by artidoro** [comment]
Thank you that seems to make it work!

---
In reply to: [488287870](https://github.com/dotnet/machinelearning/pull/3623#issuecomment-488287870) [](ancestors = 488287870)

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 11 above)
Comment (artidoro) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: I tried renaming, but that did not work (it gave an error saying that project.assets.json was not generated.
I have also tried to overwrite the Build target, but I must have made some mistake since I still found the generated .dll.

What I did was adding:
 <Target Name="Build">
    
  </Target>

to the .csproj file. 

What is the correct way to overwrite it?



---
In reply to: [280105436](https://github.com/dotnet/machinelearning/pull/3623#discussion_r280105436) [](ancestors = 280105436)

**Comment by ericstj** [comment]
failing on linux because dotnet is not on the path and the $(ToolHostCmd) is not set to point to it.

**Comment by artidoro** [comment]
Thank you @ericstj  and @eerhardt for looking at the PR, I have addressed your comments.
Let me know if there are other things I should look into!

**Comment by ericstj** [comment]
Cool, this is looking pretty good.  You may want to wait for my changes https://github.com/dotnet/arcade/pull/2672 which help simplify some of this.

**Comment by artidoro** [comment]
I would prefer checking in this change as soon as possible and updating it with the new buyer as soon as it is available.

Since we have released yesterday, it would be great to have the API Compat tool in the repo.

File: Microsoft.ML.sln [hunk 13]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27130.2026
+# Visual Studio Version 16
```
Comment (eerhardt) [inline Microsoft.ML.sln]: You may want to revert these two lines. I don't know what happens if someone tries opening the solution with VS 2017 (which is version 15) and this .sln says it is for VS 2019.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 11 above)
Comment (eerhardt) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Yes, I think that is fine.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+  <PropertyGroup>
+    <!-- needs to contain all frameworks which src projects wish to restore -->
+    <TargetFramework Condition="'$(UseIntrinsics)' != 'true'">netstandard2.0</TargetFramework>
+    <TargetFrameworks Condition="'$(UseIntrinsics)' == 'true'">netstandard2.0;netcoreapp3.0</TargetFrameworks>
```
Comment (ericstj) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Why should we have the condition here?  Is there any harm in always resolving for both TFMs?

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    <ItemGroup>
+      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
+      <_allReferenceDirectories Include="%(ReferencePath.RootDir)%(ReferencePath.Directory)" />
+      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
+      <_contractReferencePath DependencyPaths="@(_allReferenceDirectories)" />
+    </ItemGroup>
+    <Error Condition="'@(_contractReferencePath)' == ''" Text="Could not locate $(ContractName)" />
+  </Target>
+
+  <Target Name="Build">
```
Comment (ericstj) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: This could be improved by changing the project extension and defining a couple targets.  We do that elsewhere: 
https://github.com/dotnet/project-system/issues/4647

IOW: call this a .proj, or a .restoreproj or something.  SLN entry looks the same (you have to do it manually in text editor, IDE won't let you).  Add the workaround I linked, define stub targets for anything that fails, and it should avoid 
the confusion around a CSProj that doesn't build anything.
You don't need to do that now, but I think it would make this more sensible.

**Comment by ericstj** [comment]
Have you made sure you see a failure when making a breaking API change?

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 14 above)
Comment (eerhardt) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Usually `netcoreapp3.0` causes problems when you are building in VS 2017 where the SDK throws an error saying `I don't support netcoreapp3.0`.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 15 above)
Comment (artidoro) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: I have tried to add the lines found in the issue to the new .restoreproj file but that did not work. 
I will keep it as is for now then.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj (hunk 14 above)
Comment (ericstj) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: I see, makes sense.

**Comment by artidoro** [comment]
Yes I have done a few tests where I rename methods and such and the build was failing.

File: tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+  </ItemGroup>
+
+  <!-- The purpose of this target is to return a path from a referenced
+       package / project -->
+  <Target Name="GetContract" DependsOnTargets="ResolveReferences" Returns="@(_contractReferencePath)">
+    <Error Condition="'$(ContractName)' == ''" Text="ContractName must be specified" />
+    <ItemGroup>
+      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
+      <_allReferenceDirectories Include="%(ReferencePath.RootDir)%(ReferencePath.Directory)" />
+      <_contractReferencePath Include="@(ReferencePath)" Condition="'%(FileName)' == '$(ContractName)'" />
```
Comment (RussKie) [inline tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: Lines 24 and 26 are identical... 😕

Comment (eerhardt) [reply tools-local/Microsoft.ML.StableApi/Microsoft.ML.StableApi.csproj]: @RussKie - can you log an issue (or even submit a PR for the fix)?
